### ISSUE TITLE
instanceof Function broken inside of sandboxed module

### ISSUE THREAD

**Issue by OliverJAsh** [felixge/node-sandboxed-module#44]
I have a sandboxed module with the following contents:

``` js
var foo = function () {}
console.log(1, typeof foo);
console.log(2, foo instanceof Function);
// => 1 'function'
// => 2 false
```

`foo instanceof Function` should return `true`, but it returns `false`.

**Comment by joshperry** [comment]
This actually causes more subtle issues. Here's another example:

sandboxed.js
```
var defaults = { foo: 'bar' }
exports.dostuff = function dostuff(options) {
  if(!options) {
    options = {};
  }

  options = _.merge(options, defaults);

  return options;
}
```

test.js
```
it('should use defaults with no params', function() {
  var opts = mod.dostuff();
  opts.should.contain({ foo: 'bar' });
});
```

This throws an error because `should` is not on the object created in the sandboxed module.

**Comment by domenic** [comment]
@joshperry that's basically the definition of sandboxing, though. You aren't affected by modifications to the environment's prototypes; instead you run in your own clean sandbox.

**Comment by joshperry** [comment]
@domenic Yeah, sorry. This was being used in another project I'm trying to clean up as a way to mock `require`d modules so I think I assumed that the lib was meant for testing. Looks like I assumed wrong. I'll change it to use proxyquire.

**Comment by elliotstokes** [comment]
I'm having this issue as well. A quick and dirty workaround is to inject the dependency - so I was getting the issue with the moment date lib.

```javascript
test = sandboxedModule.require('filetotest', {
       requires: {
        'moment' : require('moment')
     },
});
```

**Comment by joshperry** [comment]
@elliotstokes Great idea.

**Comment by tristanls** [comment]
I ran into this again, this time without being able to work around it. My code was using [Joi](https://github.com/hapijs/joi) to do some validation:
```javascript
...
  thing: Joi.object().type(Map)
...
```
and that runs again into the `instanceof` problem.

As @domenic pointed out before, this module sandboxes, and that works great for most tests as I've happily written hundreds of tests using this module. This is the only corner case I couldn't work around. 

For testing, using [proxyquire](https://github.com/thlorenz/proxyquire) resolved my `instanceof` issue.
